src/py/kaleido/__init__.py

```diff
@@ -0,0 +1,81 @@
+# don't use scopes!
+from pathlib import Path
+import asyncio
+import base64
+import json
+
+from choreographer import Browser
+
+
+script_path = Path(__file__).resolve().parent / "vendor" / "index.html"
+
+_all_formats_ = ("png", "jpg", "jpeg", "webp", "svg", "pdf", "eps", "json")
+_text_formats_ = ("svg", "json", "eps")
+
+_scope_flags_ = ("plotlyjs", "mathjax", "topojson", "mapbox_access_token")
+
+def to_image_block(spec):
+    loop = None
+    try:
+        loop = asyncio.get_running_loop()
+    except RuntimeError:
+        pass
+    if loop:
+        # TODO: create thread and post the new loop there and run the thing and return it in a message queue :-(
```

Comment:
should the `...` on line 25 be there?

Reply:
So we can't use asyncio + old kaleido API (i thought it would be easier to pop open a thread w/ no event loop running since asyncio is one loop per thread, but behavior isn't intuitive)

It is possible to use this with asyncio, but it'll have to be a 4 line "how to use this with asyncio" in the readme

Reply:
@gvwilson for visibility